### ISSUE TITLE
Add support for `relatedTarget` in `FocusEventArgs` for `MemoEdit` and `TextEdit`

### ISSUE THREAD

**Issue by Jinjinov** [Megabit/Blazorise#4554]
**Is your feature request related to a problem? Please describe.**

There are many cases where it is very useful to know the related target of the focus event of `MemoEdit` and `TextEdit`.

For example, if we want to select a parent on a `focusin` event for any of its children, and deselect the parent on `focusout` for any of its children, we must know if we clicked on any other child in the same parent or if we clicked completely outside the parent. This is where `relatedTarget` would be very useful. (Workaround is possible, but not simple)

https://learn.microsoft.com/en-us/aspnet/core/blazor/components/event-handling?view=aspnetcore-7.0#built-in-event-arguments

> Focus | FocusEventArgs | Doesn't include support for relatedTarget.

**Describe the solution you'd like**

Provide the `relatedTarget` from JavaScript `blur`, `focus`, `focusin` and `focusout` event arguments for `MemoEdit` and `TextEdit`.

https://developer.mozilla.org/en-US/docs/Web/API/FocusEvent/relatedTarget

If `relatedTarget` is not possible to implement, `relatedTarget.id` would be enough.

### Additional context

An example would be the notes in https://keep.google.com/ if the selected note stayed the same, where it was (in the current version of Google Keep the selected note comes to foreground). The note contains two child elements: a `TextEdit` for the title, and a `MemoEdit` for the body, and the note has focus if any one of the two child elements has focus.

**Comment by stsrki** [comment]
I don't think this is possible. Found this in the `FocusEventArgs`

![image](https://user-images.githubusercontent.com/900302/218561747-a0117fa1-a1e3-4722-b577-0d2b4e2ee4e6.png)

**Comment by Jinjinov** [comment]
Yes, I also found that. It would be impossible to include `relatedTarget` for every element in then DOM, because you would need `@ref` and `ElementReference` for all of them.

But perhaps if the HTML `id` is set only for `MemoEdit` and `TextEdit`, it is possible to get the `id` in the `focusin` and `focusout` events for `relatedTarget` - I guess the logic for that would have to be inside JS. Or there is some other way to know that the user clicked inside the same parent - perhaps JS could return a `bool` if the `relatedTarget` is inside the same parent.

If `child 1` (TextEdit) has focus and I click in `child 2` (MemoEdit) then `parent 1` must remain in edit mode.

If I click outside of `parent 1`, then `child 1` and `child 2` must change to read-only.

The problem is that if I do this in `FocusIn` and `FocusOut` events, TextEdit loses focus first, then MemoEdit gains focus - and parent flickers for a second because there is a moment when there both TextEdit and MemoEdit don't have focus.

Now I have a workaround, but it is very ugly and doesn't allow me to have all the features I want.

![image](https://user-images.githubusercontent.com/20396198/218567584-073533a1-b3fc-4278-a79d-45244df81942.png)
